### PR TITLE
FIX: Use correct group out of multiple for SMTP sender

### PR DESCRIPTION
**Pull request by martin-brennan**
When there are multiple groups on a topic, we were selecting
the first from the topic allowed groups to act as the sender
email address when sending group SMTP replies via PostAlerter.
However, this was not ordered, and since there is no created_at
column on TopicAllowedGroup we cannot order this nicely, which
caused just a random group to be used (based on whatever postgres
decided it felt like that morning).

This commit changes the group used for SMTP sending to be the
group using the email_username of the to address of the first
incoming email for the topic, if there are more than one allowed
groups on the topic. Otherwise it just uses the only SMTP enabled
group.

### CODE REVIEW

File: spec/services/post_alerter_spec.rb [hunk 1]
```diff
@@ -1374,6 +1374,21 @@ def create_post_with_incoming
...
+    it "sends a group smtp email when the original group has had SMTP disabled and there is an additional topic allowed group" do
+      incoming_email_post = create_post_with_incoming
+      topic = incoming_email_post.topic
+      other_allowed_group = Fabricate(:smtp_group)
+      TopicAllowedGroup.create(group: other_allowed_group, topic: topic)
+      post = Fabricate(:post, topic: topic)
+      group.update!(smtp_enabled: false)
+      expect { PostAlerter.new.after_save_post(post, true) }.to change { ActionMailer::Base.deliveries.size }.by(1)
+      email = ActionMailer::Base.deliveries.last
+      expect(email.from).to include(other_allowed_group.email_username)
```
Comment (CvX) [inline spec/services/post_alerter_spec.rb:1386]: May I suggest some whitespace to slightly improve the code soup? 😉 

```suggestion

      expect { PostAlerter.new.after_save_post(post, true) }.to change { ActionMailer::Base.deliveries.size }.by(1)

      email = ActionMailer::Base.deliveries.last
      expect(email.from).to include(other_allowed_group.email_username)
```

File: app/services/post_alerter.rb [hunk 2]
```diff
@@ -640,7 +640,14 @@ def notify_pm_users(post, reply_to_user, notified)
 
   def group_notifying_via_smtp(post)
     return nil if !SiteSetting.enable_smtp || post.post_type != Post.types[:regular]
-    post.topic.allowed_groups.where(smtp_enabled: true).first
+    if post.topic.allowed_groups.count == 1
+      return post.topic.first_smtp_enabled_group
+    end
+    group = Group.find_by_email(post.topic.incoming_email.first.to_addresses)
+    if group.blank? || !group.smtp_enabled
```
Comment (CvX) [inline app/services/post_alerter.rb:647]:
```suggestion

    group = Group.find_by_email(post.topic.incoming_email.first.to_addresses)
    if !group&.smtp_enabled
```

**Review by CvX — APPROVED** [review]

**Comment by martin-brennan** [comment]
Thanks @CvX , thinned out the soup now ;)

File: app/services/post_alerter.rb [hunk 3]
```diff
@@ -640,7 +640,15 @@ def notify_pm_users(post, reply_to_user, notified)
 
   def group_notifying_via_smtp(post)
     return nil if !SiteSetting.enable_smtp || post.post_type != Post.types[:regular]
-    post.topic.allowed_groups.where(smtp_enabled: true).first
+    if post.topic.allowed_groups.count == 1
+      return post.topic.first_smtp_enabled_group
+    end
+
+    group = Group.find_by_email(post.topic.incoming_email.first.to_addresses)
```
Comment (CvX) [inline app/services/post_alerter.rb:647]: @martin-brennan Looking at the error logs: `post.topic.incoming_email.first` can be nil

Comment (martin-brennan) [reply app/services/post_alerter.rb:647]: https://github.com/discourse/discourse/pull/14981

**Pull request merged**